app/2.1.x/plugin-development/custom-logic.md

```diff
@@ -38,10 +38,17 @@ of the execution life-cycle of Kong:
 | `:certificate()`   | [ssl_certificate] | Executed during the SSL certificate serving phase of the SSL handshake.
 | `:rewrite()`       | [rewrite]         | Executed for every request upon its reception from a client as a rewrite phase handler. *NOTE* in this phase neither the `Service` nor the `Consumer` have been identified, hence this handler will only be executed if the plugin was configured as a global plugin!
 | `:access()`        | [access]          | Executed for every request from a client and before it is being proxied to the upstream service.
+| `:response()` | [access] | Replaces both `header_filter()` and `body_filter()`.  Executed after the entire response has been received from the Upstream service, but before sending any part of the response to the client.
```

Comment:
Should this link to a response phase doc?

Reply:
the phases linked by the other methods are OpenResty phases.  there's no OpenResty "response" phase, the `:response()` method is managed during the OpenResty "access" phase, using the buffered proxy mechanism (that handles the upstream during access)